doc/nrf/protocols/bt/bt_mesh/configuring.rst

```diff
@@ -172,3 +172,25 @@ The following configuration options are relevant when using the LPN feature:
 * Reducing the Node ID advertisement timeout decreases the period where the device consumes power for advertising.
 
   * :kconfig:option:`CONFIG_BT_MESH_NODE_ID_TIMEOUT` =30.
+
+Persistent storage
+------------------
+
+Zephyr's Bluetooth mesh has been designed to use the :ref:`zephyr:settings_api` Settings subsystem to store internal states and options in the :ref:`zephyr:bluetooth_mesh_persistent_storage` persistent memory.
+The Settings subsystem can be used with different backends. Mesh is configured with :ref:`zephyr:nvs_api` NVS as the Settings backend.
+However, in general, the Settings subsystem based on NVS is not well designed to store a huge number of states simultaneously
```

Comment:
I'd avoid saying that something is not well designed. I'd recommend just to say that when many entries are stored through the settings subsystem (not necessarily RPL), the store time grows significantly. And that users can experience this when using huge RPL.